nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-site-to-site/src/main/java/org/apache/nifi/remote/PeerDescriptionModifier.java

```diff
@@ -0,0 +1,160 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.remote;
+
+import org.apache.nifi.attribute.expression.language.PreparedQuery;
+import org.apache.nifi.attribute.expression.language.Query;
+import org.apache.nifi.remote.protocol.SiteToSiteTransportProtocol;
+import org.apache.nifi.util.NiFiProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+public class PeerDescriptionModifier {
+
+    private static final Logger logger = LoggerFactory.getLogger(PeerDescriptionModifier.class);
+
+    public enum RequestType {
+        SiteToSiteDetail,
+        Peers
+    }
+
+    private static class Route {
+        private String name;
+        private SiteToSiteTransportProtocol protocol;
+        private PreparedQuery predicate;
+        private PreparedQuery hostname;
+        private PreparedQuery port;
+        private PreparedQuery secure;
+
+        private boolean isValid() {
+            if (hostname == null) {
+                logger.warn("Ignore invalid route definition {} because 'hostname' is not specified.", name);return false;
+            }
+            if (port == null) {
+                logger.warn("Ignore invalid route definition {} because 'port' is not specified.", name);
+                return false;
+            }
+            return true;
+        }
+
+        private PeerDescription getTarget(final Map<String, String> variables) {
+            final String targetHostName = hostname.evaluateExpressions(variables, null);
+            if (isBlank(targetHostName)) {
+                throw new IllegalStateException("Target hostname was not resolved for the route definition " + name);
+            }
+
+            final String targetPortStr = port.evaluateExpressions(variables, null);
+            if (isBlank(targetPortStr)) {
+                throw new IllegalStateException("Target port was not resolved for the route definition " + name);
+            }
+
+            String targetIsSecure = secure == null ? null : secure.evaluateExpressions(variables, null);
+            if (isBlank(targetIsSecure)) {
+                targetIsSecure = "false";
```

Comment:
@mcgilman The blank check is not needed since Boolean.valueOf returns `false` for null/empty string. I will remove these lines and also callout that default is 'false' in the docs.